addons/web/static/src/boot.js

```diff
@@ -440,4 +430,62 @@
         // visit each root to find cycles
         return visitJobs(jobs.map((j) => j.name));
     }
+
+    function displayModuleErrors({ failed, missing, unloaded, cycle }) {
+        const list = (heading, arr) => {
```

Comment:
Since you're already creating a helper, maybe a second one would be feasible to reclaim some readability?
```javascript
function h(tag, ...children) {
     const el = document.createElement(tag);
     for (const child of children) {
         el.appendChild(child instanceof Node ? child : new Text(child));
     }
     return el;
}
```
Maybe reserving the second positional parameter for attributes as well.

Reply:
I tried this and found it was even harder to read. It's not horrible as is so I think I'll just just keep it like that.